### PR TITLE
Fix - Don't download EDF data by channel

### PR DESCRIPTION
**Pull request by dkeden**
Due to the way EDFs are structured and seer-py download function, data downloaded from S3 is reshaped based on the number of channel groups (i.e. column names).

By supplying a slice of the metadata, data is incorrectly reshaped.

This fix is a workaround-ish. Download data, then slice.

Could someone please pull this branch and do a sanity check - i.e. values are different between Empatica ACC channels? @EwanNurse @cfranklin11

### CODE REVIEW

**Comment by EwanNurse** [comment]
The data downloaded from the Empatica ACC channel group is now different for each channel, working as intended. Let's hold off merging until we get the green-light from Rad just to be super cautious
image

**Review by EwanNurse — APPROVED** [review]
LGTM! Works as intended. Let's just hold off merging until the db migration is done.

File: Examples/msg_data_downloader.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def run(client, path_out):
     if not isdir(output_dir):
         mkdir(output_dir)
 
-    download_status_file = get_download_status_file(client)
+    download_status_file = get_download_status_file()
```
Comment (cfranklin11) [inline Examples/msg_data_downloader.py:46]: This doesn't seem related to the other changes, just want to make sure it's intentional

**Review by cfranklin11 — APPROVED** [review]
Makes sense. Just some optional syntax suggestions

File: Examples/downloader/downloader.py [hunk 2]
```diff
@@ -18,33 +18,42 @@ def get_download_log_file():
...
 
     def get_label_groups(self):
         return self.client.get_label_groups_for_studies([self.study_id])
 
     def get_channels(self):
-        channels = self.channel_groups[['channelGroups.name', 'channels.name']].drop_duplicates()
-        return channels.to_numpy().tolist()
+        channels = self.study_metadata[['channelGroups.name', 'channels.name']].drop_duplicates()
+        return ([channel_group, channel] for channel_group, channel in zip(
+            channels['channelGroups.name'].tolist(), channels['channels.name'].tolist()))
```
[the remark this answers is not in the document]
Comment (norrishd) [reply Examples/downloader/downloader.py:43]: `iterrows()` is the slowest way to iterate over a dataframe, and `zip()` is the fastest:
https://stackoverflow.com/a/55557758

The only thing I would suggest is that I don't think it's necessary to call `tolist()` on the series?

File: Examples/msg_data_downloader.py (hunk 1 above)
Comment (dkeden) [reply Examples/msg_data_downloader.py:46]: You're right, it's not. It just bothered me, and I figured that this repo is slowly becoming a desolate place so it doesn't matter too much 😄

File: Examples/downloader/downloader.py [hunk 3]
```diff
@@ -18,33 +18,42 @@ def get_download_log_file():
...
-            segment_file = join(
-                folder_out, f'{self.study_name}_{channel_group}_{channel}_segment_{index}.parquet')
-            if isfile(segment_file):
+            segment_metadata = channel_group_metadata[channel_group_metadata['segments.id'] ==
+                                                      segment_id]
+            segment_files_list = []
+            for channel in channels:
+                segment_files_list.append(
+                    join(folder_out,
+                         f'{self.study_name}_{channel_group}_{channel}_segment_{index}.parquet'))
```
[the remark this answers is not in the document]
Comment (dkeden) [reply Examples/downloader/downloader.py:54]: Agreed. Great, thank you I'll add in

File: Examples/downloader/downloader.py [hunk 4]
```diff
@@ -54,43 +63,53 @@ def get_segment_data(self, folder_out, channel_group, channel, channel_metadata,
...
+
+            # Get channels
+            channels = channel_group_metadata['channels.name'].unique()
             # Save channel metadata
-            channel_metadata.to_csv(channel_metadata_file)
+            for channel in channels:
+                # Get channel group metadata
+                channel_metadata = self.study_metadata[
+                    (self.study_metadata['channelGroups.name'] == channel_group)
+                    & (self.study_metadata['channels.name'] == channel)]
```
[the remark this answers is not in the document]
Comment (dkeden) [reply Examples/downloader/downloader.py:98]: Absolutely. Thanks!

File: Examples/downloader/downloader.py [hunk 5]
```diff
@@ -54,43 +63,53 @@ def get_segment_data(self, folder_out, channel_group, channel, channel_metadata,
...
                 continue
 
-            segment_row = segment_metadata.iloc[0, :]
+            raw_data_list = []
+            for channel in channels:
+                raw_data_list.append(
+                    pd.DataFrame({
+                        'time': segment_data['time'].tolist(),
+                        'data': segment_data[channel].tolist()
+                    }))
```
[the remark this answers is not in the document]
Comment (dkeden) [reply Examples/downloader/downloader.py:72]: Great idea

**Review by norrishd — APPROVED** [review]
Weehoo! ⚡ 🚀 🎸
